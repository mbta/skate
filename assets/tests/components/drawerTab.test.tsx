import { jest, describe, test, expect } from "@jest/globals"
import { render } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import React from "react"
import renderer from "react-test-renderer"

import DrawerTab from "../../src/components/drawerTab"

describe("drawerTab", () => {
  test("has correct icon when visible", () => {
    const tree = renderer.create(
      <DrawerTab isVisible={true} toggleVisibility={jest.fn()} />
    )

    expect(tree).toMatchSnapshot()
  })

  test("has correct icon when collapsed", () => {
    const tree = renderer.create(
      <DrawerTab isVisible={false} toggleVisibility={jest.fn()} />
    )

    expect(tree).toMatchSnapshot()
  })

  test("calls callback when clicked", async () => {
    const mockCallback = jest.fn()
    const result = render(
      <DrawerTab isVisible={false} toggleVisibility={mockCallback} />
    )

    await userEvent.click(result.getByRole("button"))
    expect(mockCallback).toHaveBeenCalled()
  })
})
