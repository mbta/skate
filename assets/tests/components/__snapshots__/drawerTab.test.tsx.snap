// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawerTab has correct icon when collapsed 1`] = `
<div
  className="c-drawer-tab"
>
  <button
    className="c-drawer-tab__tab-button"
    data-testid="drawer-tab-button"
    onClick={[MockFunction]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg/>",
        }
      }
    />
  </button>
</div>
`;

exports[`drawerTab has correct icon when visible 1`] = `
<div
  className="c-drawer-tab"
>
  <button
    className="c-drawer-tab__tab-button"
    data-testid="drawer-tab-button"
    onClick={[MockFunction]}
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg/>",
        }
      }
    />
  </button>
</div>
`;
